File: Sliding Window Algorithms/LeetCode 0209.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        left = 0
        right = 0
        total = 0
        ans = float('inf')
        n = len(nums)

        while right < n:
            total += nums[right]
            right += 1
            while total >= target:
                ans = min(ans, right - left)
                total -= nums[left]
                left += 1

        return 0 if ans == float('inf') else ans

File: Sliding Window Algorithms/LeetCode 0209.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int right = 0;
        int sum = 0;
        int ans = Integer.MAX_VALUE;
        int n = nums.length;
        while(right < n)
        {
            sum += nums[right++];
            while(sum >= target)
            {
                ans = Math.min(ans, right - left);
                sum -= nums[left++];
            }
        }
        if(ans == Integer.MAX_VALUE)
        {
            return 0;
        }
        return ans;
    }
}
